docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-cloud-providers/amazon.md

```diff
@@ -159,6 +163,357 @@ Setting the value of the tag to `owned` tells the cluster that all resources wit
+tolerations:
+  - effect: NoSchedule
+    key: node.cloudprovider.kubernetes.io/uninitialized
+    value: 'true'
+  - effect: NoSchedule
+    value: 'true'
+    key: node-role.kubernetes.io/controlplane
+
+```
+
+```
+nodeSelector:
+  node-role.kubernetes.io/controlplane: 'true'
+```
+
+:::note for Rancher `<v2.8.0`:
+	
+There's currently [a known issue](https://github.com/rancher/dashboard/issues/9249) which doesn't allow nodeSelector to be updated from the Rancher UI.  Continue installing the chart and then edit the daemonset manually to set the `nodeSelector`. 
+
+:::
+
+10. Install the chart and confirm that daemonset `aws-cloud-controller-manager` is running - verify `aws-cloud-controller-manager` pods are running in target namespace (`kube-system` unless modified in step 6). 
+
+```
+
+```
+
+### Migrating to the Out-of-Tree AWS Cloud Provider for RKE1
+
+In order to upgrade existing cluster with in-tree cloud provider to AWS cloud controller manager, you can run stop kube controller manager and install AWS cloud controller manager in many ways. Refer to [External cloud controller manager](https://cloud-provider-aws.sigs.k8s.io/getting_started/) for details.
+
+When downtime is acceptable, you can switch to external cloud provider which removes in-tree components and then deploy charts to install AWS cloud controller manager as explained in [using out of tree cloud provider](#using-out-of-tree-aws-cloud-provider).
+
+When control plane cannot tolerate downtime, leader migration must be enabled to facilitate a smooth migration from the controllers in the kube controller manager to their counterparts in the cloud controller manager. Refer to [Using leader migration](https://cloud-provider-aws.sigs.k8s.io/getting_started/) for more details.
+
+**Important**
+- [Cloud controller migration docs](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#before-you-begin) mentions that it is possible to migrate with the same Kubernetes version, but assumes that migration is part of Kubernetes upgrade.
+
+- Refer [Migrate to use cloud controller manager](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/) to confirm if any customizations are required before migrating.
+  Confirm [migration configuration values](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#default-configuration) and special case around [migrating IPAM controllers](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#node-ipam-controller-migration).
+
+
+1. Update cluster config to enable leader migration in `cluster.yml`
+
+```
+services:
+  kube-controller:
+    extra_args:
+      enable-leader-migration: "true"
+```
+Note that cloud provider is still `aws` at this step.
+```
+cloud_provider:
+  name: aws
+```
+
+2. Cordon control plane nodes so aws cloud controller pods run on nodes only after upgrading to external cloud provider.
```

Comment:
```suggestion
2. Cordon the control plane nodes, so that AWS cloud controller pods run on nodes only after upgrading to the external cloud provider.
```